spec/features/user/user_can_invite_to_app_spec.rb

```diff
@@ -0,0 +1,124 @@
+require 'rails_helper'
+
+RSpec.describe 'User can send an email invite to the application' do
+  describe 'As a logged in user, my dashboard' do
+    before :each do
+      @user = create(:user)
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+
+      stub_user_1_dashboard
+
+      ActionMailer::Base.deliveries = []
+
+    end
+    describe 'if I am connected to Github' do
+      before :each do
+        create(:github_token, user: @user, token: ENV['USER_1_GITHUB_TOKEN'])
+      end
+
+      it 'has a Send an Invite button' do
+
+        visit dashboard_path
+
+        click_button("Send an Invite")
+
+        expect(current_path).to eq(invite_path)
+
+      end
+
+      describe 'When I click on send an Invite'  do
+        it 'my path is now /invite' do
+
+          visit dashboard_path
+
+          click_button("Send an Invite")
+
+          expect(current_path).to eq('/invite')
+
+        end
+
+        describe 'if I enter a valid github handle' do
+          it 'should send an email to that github user inviting them to the app' do
+            visit dashboard_path
+
+            click_button("Send an Invite")
+
+            expect(current_path).to eq('/invite')
+
+            fill_in 'handle', with: 'teresa-m-knowles'
+
+            stub_get_json("https://api.github.com/users/teresa-m-knowles", "user_1_github_get_user.json")
+
+            click_button("Send Invite")
+
+            expect(page).to have_content("Successfully sent invite!")
+
+            email = ActionMailer::Base.deliveries.last
+            email_body = email.part.first.body.raw_source
+            
+            expect(email.subject).to eq("#{@user.first_name} wants to invite you to Brownfield of Dreams.")
+
+            expect(email_body).to have_link("Register", href: "http://localhost/register")
+            expect(ActionMailer::Base.deliveries.count).to eq(1)
```

Comment:
I'd check/maybe delete this last test. For me, the count changed depending on running just this test or the whole suite.

Reply:
Yes, I noticed that too. I added 

```
ActionMailer::Base.deliveries = []
```

to the before_each setup to wipe them. Another option is to expect the count of deliveries to change by 1.

Reply:
Ah. Good idea.